### DISCUSSION TITLE
data lose

### DISCUSSION THREAD

**Discussion by taozix** [juicedata/juicefs#3636] [General]
**What happened**:
data lost

**What you expected to happen**:
data consistent

**How to reproduce it (as minimally and precisely as possible)**:
juicefs mount redis://xxx /test
juicefs gateway :9000

1.mkdir /test/abc && cd /test/abc && echo "aaa" >a.txt
2.login juicefs gateway and delete obj a.txt
3.echo "bbb" >b.txt && ls
4.No obj b.txt in juicefs gateway
5.cd .. 
6.No dir abc in juicefs mount point.

**Anything else we need to know?**

**Environment**:
- juicefs version 1.0.4+2023-04-06.f1c475d

**Comment by zhijian-pro** [comment]
This is expected because there is no concept of directories in the s3 protocol. If the '/test/abc/' object does not exist, and no object prefixed with '/test/abc/' exists, then '/test/abc/' should not exist
